docs/CODE_GUIDELINES.md

```diff
@@ -232,6 +232,22 @@ Use:
 }
 ```
 
+- Avoid using classes on one component to alter the styling of a different component. For example, instead of:
+
+```css
+.banner__icon {
+  .banner--success & {
+    fill: var(--eds-theme-color-utility-success-foreground);
+  }
+}
+```
+
+Use:
+
+```tsx
+<Icon color={EdsThemeColorUtilitySuccessForeground} />
```

Comment:
May be worth having some sort of a check in here since the example CSS is targeting a specific variant, or maybe just drop the `banner--success` part in the example styles.
```suggestion
<Icon color={variant === 'success' && EdsThemeColorUtilitySuccessForeground} />
```